### ISSUE TITLE
send() method with options for file generates error

### ISSUE THREAD

**Issue by dayxhep** [hydrabolt/discord.js#1591]
**Please describe the problem you are having in as much detail as possible:**

Using 
```js
User.send(text, {file: 'file/path.jpg'});
// or
User.send(text, {files: ['file/path.jpg']});
```
generates an error.

```
TypeError: Cannot assign to read only property 'name' of function 'function (start,end) {
     if(!end){ end=-1;}
    return this.slice(start, this.length+1-(end*-1))
}'
    at User.send (.../node_modules/discord.js/src/structures/interfaces/TextBasedChannel.js:100:23)
```
The rest of the stack is the msg.author.send() calling point.
Yes the bot has rights to manipulate the file (fs can create / move / delete this file)

**Include a reproducible code sample here, if possible:**
```js
msg.author.send(
    "test", 
    {file:`./res/imgs/emb-1.png`}
).catch(throwErr)
```
msg is a Message, provided by client.on('message', msg => {...})

**Further details:**

- discord.js version: 11.1.0
- node.js version: v6.9.5
- Operating system: Linux

**Comment by SpaceEEC** [comment]
This is caused by your prototypes and has been fixed in #1461.
You could install the 11.1-dev branch as a solution.
Or maybe you can define your prototypes non enumberable or something.

**Comment by dayxhep** [comment]
Alright thanks for the explanations :)
